model/Bom.js

```diff
@@ -74,20 +75,26 @@ class Bom extends CycloneDXObject {
     // read-installed with default options marks devDependencies as extraneous
     // if a package is marked as extraneous, do not include it as a component
     if (pkg.extraneous) return
+    let component
     if (!isRootPkg) {
-      const component = new Component(pkg, this.includeLicenseText, lockfile)
+      component = new Component(pkg, this.includeLicenseText, lockfile)
       if (component.externalReferences === undefined || component.externalReferences.length === 0) {
         delete component.externalReferences
       }
       if (list[component.purl]) return // remove cycles
       list[component.purl] = component
     }
     if (pkg.dependencies) {
-      Object.keys(pkg.dependencies)
+      const deps = Object.keys(pkg.dependencies)
         .map(x => pkg.dependencies[x])
         .filter(x => typeof (x) !== 'string') // remove cycles
         .map(x => this.createComponent(x, list, lockfile))
+        .filter(x => !!x)
+      if (component && deps.length > 0) {
+        this.addDependency(new Dependency(component.bomRef, deps.map(({ bomRef }) => ({ ref: bomRef }))))
```

Comment:
`Component.bomRef` is defined as `string|undefined`

according to the spec https://github.com/CycloneDX/specification/blob/master/schema/bom-1.3-strict.schema.json#L855
the Dependency `ref` must be string - not `undefined`

Therefore an unhandled behaviour may happen,
if `component.bomRef` is unset, empty, etc ... 


please add proper JSDoc typrhints to `Dependency` to make clear which tipes to expect.

Reply:
Solved by filtering out any `undefined`.